packages/protocol/test/governance/vesting.ts

```diff
@@ -0,0 +1,1734 @@
+import { CeloContractName } from '@celo/protocol/lib/registry-utils'
+import { getParsedSignatureOfAddress } from '@celo/protocol/lib/signing-utils'
+import {
+  assertEqualBN,
+  assertLogMatches,
+  assertRevert,
+  NULL_ADDRESS,
+  timeTravel,
+} from '@celo/protocol/lib/test-utils'
+import { BigNumber } from 'bignumber.js'
+import * as _ from 'lodash'
+import { upperFirst } from 'lodash'
+import {
+  AccountsContract,
+  AccountsInstance,
+  GoldTokenContract,
+  GoldTokenInstance,
+  LockedGoldContract,
+  LockedGoldInstance,
+  MockElectionContract,
+  MockElectionInstance,
+  MockGovernanceContract,
+  MockGovernanceInstance,
+  MockValidatorsContract,
+  MockValidatorsInstance,
+  RegistryContract,
+  RegistryInstance,
+  VestingFactoryContract,
+  VestingFactoryInstance,
+  VestingInstanceContract,
+} from 'types'
+import Web3 = require('web3')
+
+const ONE_GOLDTOKEN = new BigNumber('1000000000000000000')
+
+const authorizationTests: any = {}
+const authorizationTestDescriptions = {
+  voting: {
+    me: 'vote signing key',
+    subject: 'voteSigner',
+  },
+}
+
+interface IVestingSchedule {
+  vestingBeneficiary: string
+  vestingNumPeriods: number
+  vestingCliff: number
+  vestingStartTime: number
+  vestingPeriodSec: number
+  vestAmountPerPeriod: BigNumber
+  vestingRevocable: boolean
+  vestingRevoker: string
+  vestingMaxPausePeriod: number
+}
+
+const VestingFactory: VestingFactoryContract = artifacts.require('VestingFactory')
+const VestingInstance: VestingInstanceContract = artifacts.require('VestingInstance')
+const Accounts: AccountsContract = artifacts.require('Accounts')
+const LockedGold: LockedGoldContract = artifacts.require('LockedGold')
+const GoldToken: GoldTokenContract = artifacts.require('GoldToken')
+const MockGovernance: MockGovernanceContract = artifacts.require('MockGovernance')
+const MockValidators: MockValidatorsContract = artifacts.require('MockValidators')
+const MockElection: MockElectionContract = artifacts.require('MockElection')
+const Registry: RegistryContract = artifacts.require('Registry')
+
+// @ts-ignore
+// TODO(mcortesi): Use BN
+LockedGold.numberFormat = 'BigNumber'
+
+const MINUTE = 60
+const HOUR = 60 * 60
+const DAY = 24 * HOUR
+const MONTH = 30 * DAY
+const UNLOCKING_PERIOD = 3 * DAY
+const MAX_PAUSE_PERIOD = 365 * DAY
+
+contract('Vesting', (accounts: string[]) => {
+  const owner = accounts[0]
+  const beneficiary = accounts[1]
+  const revoker = accounts[2]
+  let accountsInstance: AccountsInstance
+  let lockedGoldInstance: LockedGoldInstance
+  let goldTokenInstance: GoldTokenInstance
+  let vestingFactoryInstance: VestingFactoryInstance
+  let mockElection: MockElectionInstance
+  let mockGovernance: MockGovernanceInstance
+  let mockValidators: MockValidatorsInstance
+  let registry: RegistryInstance
+
+  const vestingDefaultSchedule: IVestingSchedule = {
+    vestingBeneficiary: beneficiary,
+    vestingNumPeriods: 4,
+    vestingCliff: HOUR,
```

Comment:
This is a bizarre choice given the vesting period of 3 months. Consider monthly or quarterly vesting with a 1 year cliff

Reply:
this should induce changes to all unit test as they all derive from the same default scheme.